### PR TITLE
REFACTOR-#5386: move `partition.split` implementation in base class

### LINKED ISSUE #5386: REFACTOR: move `partition.split` implementation into base class

### PR DESCRIPTION
**Pull request by anmyachev**
Signed-off-by: Anatoly Myachev <anatoly.myachev@intel.com>

## What do these changes do?

- [x] first commit message and PR title follow format outlined [here](https://modin.readthedocs.io/en/latest/development/contributing.html#commit-message-formatting)
  > **_NOTE:_**  If you edit the PR title to match this format, you need to add another commit (even if it's empty) or amend your last commit for the CI job that checks the PR title to pick up the new PR title.
- [x] passes `flake8 modin/ asv_bench/benchmarks scripts/doc_checker.py`
- [x] passes `black --check modin/ asv_bench/benchmarks scripts/doc_checker.py`
- [x] signed commit with `git commit -s` 
- [x] Resolves #5386 
- [x] tests added and passing
- [x] module layout described at `docs/development/architecture.rst` is up-to-date

### CODE REVIEW

**Review by vnlitvinov — APPROVED** [review]
Cool, less code duplication! LGTM!

File: modin/core/execution/dask/implementations/pandas_on_dask/partitioning/partition.py [hunk 1]
```diff
@@ -268,30 +273,6 @@ def ip(self):
...
-            Arguments to pass to ``split_func``.
-
-        Returns
-        -------
-        list
-            A list of partitions.
-        """
-        outputs = DaskWrapper.deploy(
-            split_func, [self._data] + list(args), num_returns=num_splits, pure=True
-        )
```
Comment (dchigarev) [inline modin/core/execution/dask/implementations/pandas_on_dask/partitioning/partition.py:292]: what about the missing `pure=True` parameter in the base implementation, is it significant?

Comment (anmyachev) [reply modin/core/execution/dask/implementations/pandas_on_dask/partitioning/partition.py:292]: `pure=True` is a default value for `DaskWrapper.deploy`, so no problem

**Review by dchigarev — APPROVED** [review]

File: modin/core/execution/dask/implementations/pandas_on_dask/partitioning/partition.py [hunk 2]
```diff
@@ -40,6 +42,8 @@ class PandasOnDaskDataframePartition(PandasDataframePartition):
         Call queue that needs to be executed on wrapped pandas DataFrame.
     """
 
+    execution_wrapper = DaskWrapper
```
Comment (YarShev) [inline modin/core/execution/dask/implementations/pandas_on_dask/partitioning/partition.py:45]: I would prefer to name `engine_wrapper`, which is similar to the file names we have.

**Pull request merged**